lib/config/convict.js

```diff
@@ -0,0 +1,835 @@
+    },
+    reportURI: {
+      doc: 'Allows to add a URL for CSP reports in case of violations.',
+      format: '*',
+      default: undefined
+    }
+  },
+  protocolUseSSL: {
+    doc: 'Set to use TLS protocol for resources path (only applied when domain is set).',
+    format: 'Boolean',
+    env: 'CMD_PROTOCOL_USESSL',
+    default: false
+  },
+  useCDN: {
+    doc: 'Set to use CDN resources or not',
+    format: 'Boolean',
+    env: 'CMD_USECDN',
+    default: true
+  },
+  allowAnonymous: {
+    doc: 'Set to allow anonymous usage. (Allow guests general usage of CodiMD.)',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_ANONYMOUS',
+    default: true
+  },
+  allowAnonymousEdits: {
+    doc: 'If `allowAnonymous` is `true`: allow users to select `freely` permission, allowing guests to edit existing notes.',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_ANONYMOUS_EDITS',
+    default: false
+  },
+  allowFreeURL: {
+    doc: 'Set to allow new note creation by accessing a nonexistent note URL. This is the behavior familiar from [Etherpad](https://github.com/ether/etherpad-lite).',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_FREEURL',
+    default: false
+  },
+  forbiddenNoteIDs: {
+    doc: 'List of note IDs that can\'t be used. Main purpose is to prevent bugs due usage of paths in free-url-mode.',
+    env: 'CMD_FORBIDDEN_NOTE_IDS',
+    default: ['robots.txt', 'favicon.ico', 'api', 'build', 'css', 'docs', 'fonts', 'js', 'uploads', 'vendor', 'views']
+  },
+  defaultPermission: {
+    doc: 'Set notes default permission (only applied on signed-in users).',
+    format: function (val) {
+      if (!/(freely|editable|limited|locked|protected|private)/.test(val)) {
+        throw new Error('defaultPermission has to be set to `freely`, `editable`, `limited`, `locked`, `protected`, or `private`')
+      }
+    },
+    env: 'CMD_DEFAULT_PERMISSION',
+    default: 'editable'
+  },
+  dbURL: {
+    doc: 'Database URL',
+    format: '*',
+    default: undefined,
+    sensitive: true
+  },
+  db: {
+    host: {
+      doc: 'Database host name/IP',
+      format: '*',
+      default: undefined
+    },
+    name: {
+      doc: 'Database name',
+      format: '*',
+      default: undefined
+    }
+  },
+  // ssl path
+  sslKeyPath: {
+    doc: 'SSL private key path absolute or relative to CodiMD\'s base director (only need when you set `useSSL`)',
+    format: 'optional-path',
+    default: ''
+  },
+  sslCertPath: {
+    doc: 'SSL certificate path absolute or relative to CodiMD\'s base director (only need when you set `useSSL`)',
+    format: 'optional-path',
+    default: ''
+  },
+  sslCAPath: {
+    doc: 'SSL CA certificate path absolute or relative to CodiMD\'s base director (only need when you set `useSSL`)',
+    format: 'optional-path',
+    default: ''
+  },
+  dhParamPath: {
+    doc: 'SSL DH parameter absolute or relative to CodiMD\'s base director (only need when you set `useSSL`)',
+    format: 'optional-path',
+    default: ''
+  },
```

Comment:
might be a good time to dump these?

Reply:
Entirely? This would eliminate the need for useTLS completely. But it would also prevent secure backend setups, where the reverse-proxy is not running on the same host as the CodiMD instance.